abstractions/go/serialization/serialization_writer_factory_registry_test.go

```diff
@@ -0,0 +1,123 @@
+package serialization
+
+import (
+	assert "github.com/stretchr/testify/assert"
+	"testing"
+    "github.com/stretchr/testify/mock"
+)
+type mockSerializer struct {
```

Comment:
Is using a mocking library something we can use to reduce test code maintenance overhead?

Reply:
all mocking libraries in Go were implemented before the ability of generics which means you have to handcraft the methods of the mocks. I think we need to give them time to take advantage of generics, but for now I don't see a lot of value in adding mocking in Go unless I missed something.